### ISSUE TITLE
Box Constrained Optimization for BigFloat?

### ISSUE THREAD

**Issue by olugovoy** [JuliaNLSolvers/Optim.jl#647]
Hello,
I'm following the examples: <http://julianlsolvers.github.io/Optim.jl/stable/#user/minimization/>
Trying them for arbitrary precision numbers. The unconstrained optimization seems to work for BigFloat, but Box-constrained doesn't:

using Optim
f(x) = (1.0 - x[1])^2 + 100.0 * (x[2] - x[1]^2)^2  
x0 = ([0.0, 0.0])  
lower = ([1.25, -2.1])  
upper = ([Inf, Inf])  
initial_x = [2.0, 2.0]  
results = optimize(f, big.(initial_x))
~~~~~~~~~~~~~~~~~~~~~~
Results of Optimization Algorithm
 * Algorithm: Nelder-Mead
 * Starting Point: [2.000000000000000000000000000000000000000000000000000000000000000000000000000000,2.000000000000000000000000000000000000000000000000000000000000000000000000000000, ...]
 * Minimizer: [1.000148755487008986452378988862618787286662281460923338072838831670712558309734,1.000289855045705580486570476957652173874639251053583989362781909786941891075004, ...]
 * Minimum: 2.802345e-08
 * Iterations: 52
 * Convergence: true
   *  √(Σ(yᵢ-ȳ)²)/n < 1.0e-08: true
   * Reached Maximum Number of Iterations: false
 * Objective Calls: 102
~~~~~~~~~~~~~~~~~~~~~~~

results = optimize(f, big.(lower), big.(upper), big.(initial_x))
~~~~~~~~~~~~~~~~~~~~~~
ERROR: MethodError: no method matching initial_mu(::Array{BigFloat,1}, ::Array{BigFloat,1}, ::Float64, ::Float64)
Closest candidates are:
  initial_mu(::AbstractArray{T,N} where N, ::AbstractArray{T,N} where N) where T at /Users/OL/.julia/v0.6/Optim/src/multivariate/solvers/constrained/fminbox.jl:5
  initial_mu(::AbstractArray{T,N} where N, ::AbstractArray{T,N} where N, ::T) where T at /Users/OL/.julia/v0.6/Optim/src/multivariate/solvers/constrained/fminbox.jl:5
  initial_mu(::AbstractArray{T,N} where N, ::AbstractArray{T,N} where N, ::T, ::T) where T at /Users/OL/.julia/v0.6/Optim/src/multivariate/solvers/constrained/fminbox.jl:5
Stacktrace:
 [1] optimize(::NLSolversBase.OnceDifferentiable{BigFloat,Array{BigFloat,1},Array{BigFloat,1}}, ::Array{BigFloat,1}, ::Array{BigFloat,1}, ::Array{BigFloat,1}, ::Optim.Fminbox{Optim.LBFGS{Void,LineSearches.InitialStatic{Float64},LineSearches.HagerZhang{Float64,Base.RefValue{Bool}},Optim.##32#34},Float64,Optim.##88#90}, ::Optim.Options{Float64,Void}) at /Users/OL/.julia/v0.6/Optim/src/multivariate/solvers/constrained/fminbox.jl:209
 [2] #optimize#95(::Bool, ::Symbol, ::Function, ::Function, ::Array{BigFloat,1}, ::Array{BigFloat,1}, ::Array{BigFloat,1}, ::Optim.Fminbox{Optim.LBFGS{Void,LineSearches.InitialStatic{Float64},LineSearches.HagerZhang{Float64,Base.RefValue{Bool}},Optim.##32#34},Float64,Optim.##88#90}, ::Optim.Options{Float64,Void}) at /Users/OL/.julia/v0.6/Optim/src/multivariate/solvers/constrained/fminbox.jl:154
 [3] optimize(::Function, ::Array{BigFloat,1}, ::Array{BigFloat,1}, ::Array{BigFloat,1}) at /Users/OL/.julia/v0.6/Optim/src/multivariate/solvers/constrained/fminbox.jl:153
~~~~~~~~~~~~~~~~~~~~~~

Are there any easy fix?
It looks like some methods for BigFloat have been already defined, but not for all cases.
Or should I specify any additional parameters for the constrained optimization with BigFloat? Thx!

**Comment by olugovoy** [comment]
Trying the fix on  v0.6.4 and v0.7.0-beta2:
(Or should I use the dev version?)

julia> Pkg.clone("https://github.com/anriseth/Optim.jl/", "Optim")
julia> using Optim
~~~~~~~
[ Info: Precompiling module Optim
...
ERROR: LoadError: LoadError: UndefVarError: LineSearchResults not defined
Stacktrace:
 [1] getproperty(::Module, ::Symbol) at ./sysimg.jl:13
 [2] top-level scope at none:0
 [3] include at ./boot.jl:317 [inlined]
 [4] include_relative(::Module, ::String) at ./loading.jl:1034
 [5] include at ./sysimg.jl:29 [inlined]
 [6] include(::String) at /Users/OL/.julia/dev/Optim/src/Optim.jl:3
 [7] top-level scope at none:0
 [8] include at ./boot.jl:317 [inlined]
 [9] include_relative(::Module, ::String) at ./loading.jl:1034
 [10] include(::Module, ::String) at ./sysimg.jl:29
 [11] top-level scope at none:0
 [12] eval at ./boot.jl:319 [inlined]
 [13] eval(::Expr) at ./client.jl:394
 [14] top-level scope at ./none:3 [inlined]
 [15] top-level scope at ./<missing>:0
in expression starting at /Users/OL/.julia/dev/Optim/src/multivariate/solvers/first_order/gradient_descent.jl:39
in expression starting at /Users/OL/.julia/dev/Optim/src/Optim.jl:81
ERROR: Failed to precompile Optim to /Users/OL/.julia/compiled/v0.7/Optim/R5uo.ji.
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] macro expansion at ./logging.jl:298 [inlined]
 [3] compilecache(::Base.PkgId) at ./loading.jl:1173
 [4] _require(::Base.PkgId) at ./loading.jl:942
 [5] require(::Base.PkgId) at ./loading.jl:838
 [6] require(::Module, ::Symbol) at ./loading.jl:833
~~~~~~~~~~~

**Comment by pkofod** [comment]
It won't work on v0.7 of julia yet, but try pulling master of Optim in v0.6.4 and it should work now.
